src/combat.ts

```diff
@@ -130,52 +145,262 @@ export class Macro extends LibramMacro {
+        )
+        .externalIf(
+          myFamiliar() === $familiar`Hobo Monkey`,
+          Macro.while_(
+            `!match "shoulder, and hands you some Meat." && !pastround 20 && monsterhpabove ${passiveDamage}`,
+            Macro.item("seal tooth")
+          )
+        )
+        .if_(
+          `monsterhpabove ${passiveDamage + 40}`,
+          Macro.tryHaveItem($item`porquoise-handled sixgun`)
+        )
+    );
+  }
+
+  static meatStasis(checkPassive: boolean): Macro {
+    return new Macro().meatStasis(checkPassive);
+  }
+
+  startCombat(): Macro {
+    return this.tryHaveSkill($skill`Sing Along`)
+      .tryHaveSkill($skill`Curse of Weaksauce`)
+      .trySkill($skill`Pocket Crumbs`)
+      .trySkill($skill`Extract`)
+      .tryHaveItem($item`porquoise-handled sixgun`)
+      .externalIf(have($skill`Meteor Lore`), Macro.trySkill($skill`Micrometeorite`))
+      .tryHaveItem($item`Time-Spinner`)
+      .tryHaveItem($item`Rain-Doh indigo cup`)
+      .tryHaveItem($item`Rain-Doh blue balls`)
+      .externalIf(
+        haveEquipped($item`Buddy Bjorn`) || haveEquipped($item`Crown of Thrones`),
+        Macro.while_("!pastround 3 && !hppercentbelow 25", Macro.item("seal tooth"))
+      )
+      .externalIf(
+        myFamiliar() === $familiar`Stocking Mimic`,
+        Macro.while_("!pastround 10 && !hppercentbelow 25", Macro.item("seal tooth"))
+      )
+      .externalIf(
+        myFamiliar() === $familiar`Hobo Monkey`,
+        Macro.while_(
+          `!match "shoulder, and hands you some Meat." && !pastround 20 && !hppercentbelow 25`,
+          Macro.item("seal tooth")
+        )
+      );
+  }
+
+  static startCombat(): Macro {
+    return new Macro().startCombat();
+  }
+
+  kill(): Macro {
+    return (
+      this.externalIf(
+        myClass() === $class`Sauceror` && have($skill`Curse of Weaksauce`),
+        Macro.trySkill($skill`Curse of Weaksauce`)
+      )
+        .externalIf(
+          !(myClass() === $class`Sauceror` && have($skill`Curse of Weaksauce`)),
+          Macro.while_("!pastround 20 && !hppercentbelow 25 && !missed 1", Macro.attack())
+        )
+        // Using while_ here in case you run out of mp
+        .while_("hasskill Saucegeyser", Macro.skill($skill`Saucegeyser`))
+        .while_("hasskill Weapon of the Pastalord", Macro.skill($skill`Weapon of the Pastalord`))
+        .while_("hasskill Cannelloni Cannon", Macro.skill($skill`Cannelloni Cannon`))
+        .while_("hasskill Wave of Sauce", Macro.skill($skill`Wave of Sauce`))
+        .while_("hasskill Saucestorm", Macro.skill($skill`Saucestorm`))
+        .while_("hasskill Lunging Thrust-Smack", Macro.skill($skill`Lunging Thrust-Smack`))
+        .attack()
+        .repeat()
+    );
+  }
+
+  static kill(): Macro {
+    return new Macro().kill();
+  }
+
+  basicCombat(): Macro {
+    return this.startCombat().kill();
+  }
+
+  static basicCombat(): Macro {
+    return new Macro().basicCombat();
+  }
+
+  ghostBustin(): Macro {
```

Comment:
we use love gnats in the actual macro--could we use it to increment stunRounds as well?

Reply:
Yeah I did: `if (get("lovebugsUnlocked")) stunRounds++;`